RecoParticleFlow/PFClusterProducer/plugins/PFClusterFromHGCalMultiCluster.cc

```diff
@@ -0,0 +1,67 @@
+#include "PFClusterFromHGCalMultiCluster.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "FWCore/Framework/interface/Event.h"
+
+#include "DataFormats/ForwardDetId/interface/HGCalDetId.h"
+
+void PFClusterFromHGCalMultiCluster::updateEvent(const edm::Event& ev) {
+  ev.getByToken(_clusterToken, _clusterH);
+}
+
+void PFClusterFromHGCalMultiCluster::buildClusters(
+    const edm::Handle<reco::PFRecHitCollection>& input,
+    const std::vector<bool>& rechitMask, const std::vector<bool>& seedable,
+    reco::PFClusterCollection& output) {
+  // USE HGCal Multiclusters instead
+
+  const auto& hgcalMultiClusters = *_clusterH;
+  auto const& hits = *input;
+
+  // for quick indexing back to hit energy
+  std::unordered_map<uint32_t, size_t> detIdToIndex(hits.size());
+  for (uint32_t i = 0; i < hits.size(); ++i) {
+    detIdToIndex[hits[i].detId()] = i;
+  }
+
+  for (const auto& mcl : hgcalMultiClusters) {
+    edm::Ref<std::vector<reco::PFRecHit> > seed;
+    double energy = 0.0, highest_energy = 0.0;
+    output.emplace_back();
+    reco::PFCluster& back = output.back();
+    for (const auto& cl : mcl) {
+      const auto& hitsAndFractions = cl->hitsAndFractions();
+      for (const auto& hAndF : hitsAndFractions) {
+        auto itr = detIdToIndex.find(hAndF.first);
+        if (itr == detIdToIndex.end()) {
+	  continue;  // hit wasn't saved in reco
+	}
+        auto ref = makeRefhit(input, itr->second);
+	assert(ref->detId() == hAndF.first.rawId());
```

Comment:
Depending on your editor's setting you can end up pretty much in any situation you want.
This is such a useless use of our time (read the comment in general, not directed to you, please).
I'd urge, once again, S&C to make use of the tools we already have in our toolchain (clang-format).